Description: Allow -display option with no argument.
 Previously, if a user ran wmitime with the -display option and no argument,
 a segmentation fault would occur.  This patch prints a warning instead.
Author: Doug Torrance <dtorrance@monmouthcollege.edu>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=716466
Last-Update: 2014-08-04

--- a/wmgeneral/wmgeneral.c
+++ b/wmgeneral/wmgeneral.c
@@ -385,7 +385,11 @@
 	for (i=1; argv[i]; i++) {
 		if (!strcmp(argv[i], "-display")) {
 			display_name = argv[i+1];
-			i++;
+			if (!display_name)
+				printf("Please provide an argument for "
+				       "-display.\n");
+			else
+				i++;
 		}
 		if (!strcmp(argv[i], "-geometry")) {
 			geometry = argv[i+1];
